tensorflow/compiler/xla/service/hlo_opcode.cc

```diff
@@ -31,6 +31,24 @@ std::string HloOpcodeString(HloOpcode opcode) {
   }
 }
 
+std::string HloOpcodeString(absl::InlinedVector<HloOpcode, 1> opcodes) {
```

Comment:
absl::Span would be the appropriate data type here, instead of InlinedVector<HloOpcode, 1>.

But also you shouldn't need this function at all.  Instead you just want to call absl::StrJoin inside pattern_matcher.

Reply:
> But also you shouldn't need this function at all.

Still seems like we have this function?

Reply:
We want to join the strings returned by `HloOpcodeString(HloOpcode)`, not the `HloOpcode`s themselves.

Reply:
absl::StrJoin takes an optional lambda as its third argument which lets you do that.  (Try `git grep`'ing for some examples in the codebase.)